File: tests/fixtures/csharp/Catalog.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using F = System.IO.File;
using static System.Math;

namespace Acme.Store
{
    public interface IDescribable
    {
        string Describe();
    }

    public class Catalog : IDescribable
    {
        private readonly List<Book> books = new List<Book>();
        private int capacity;

        public Catalog(int capacity)
        {
            this.capacity = capacity;
            Console.WriteLine("catalog created");
        }

        public Catalog() : this(16)
        {
        }

        public int Count() => books.Count;

        public static Catalog WithDefaults()
        {
            var catalog = new Catalog(8);
            catalog.AddBook(new Book("Dune", 1965));
            return catalog;
        }

        public void AddBook(Book book)
        {
            if (books.Count >= capacity)
            {
                Grow();
            }
            books.Add(book);
        }

        public void AddBook(string title, int year)
        {
            this.AddBook(new Book(title, year));
        }

        private void Grow()
        {
            capacity = Max(capacity * 2, 16);
        }

        public string Describe()
        {
            var sb = new StringBuilder();
            foreach (var book in books)
            {
                sb.Append(book.Label());
            }
            var text = sb.ToString().Trim();
            Console.WriteLine(text);
            return text;
        }

        public async Task<int> LoadAsync(string path)
        {
            var text = await F.ReadAllTextAsync(path);
            int CountLines(string s) => s.Split('\n').Length;
            return CountLines(text);
        }

        public List<string> ModernTitles()
        {
            var titles = new List<string>();
            for (int i = 0; i < books.Count; i++)
            {
                var book = books[i];
                titles.Add(book.IsModern() ? book.Label() : "classic");
            }
            Console.WriteLine(this.First<string>(titles));
            return titles;
        }

        public T First<T>(List<T> items)
        {
            if (items.Count == 0)
            {
                throw new InvalidOperationException("empty");
            }
            return items[0];
        }
    }

    public record Book(string Title, int Year)
    {
        public string Label() => $"{Title} ({Year})";

        public bool IsModern()
        {
            return Year >= 1950;
        }
    }
}

File: tests/fixtures/java/Catalog.java
package com.example.library;

import java.util.ArrayList;
import java.util.List;
import java.util.function.*;
import static java.lang.Math.max;

public class Catalog {
    private final List<Book> books = new ArrayList<>();
    private int capacity;

    public Catalog() {
        this(16);
    }

    public Catalog(int capacity) {
        this.capacity = capacity;
    }

    public void addBook(Book book) {
        if (books.size() >= capacity) {
            grow();
        }
        books.add(book);
    }

    public void addBook(String title, int year) {
        this.addBook(new Book(title, year));
    }

    private void grow() {
        capacity = max(capacity * 2, 32);
    }

    public static Catalog withDefaults() {
        Catalog catalog = new Catalog();
        catalog.addBook("Dune", 1965);
        return catalog;
    }

    public <T extends Comparable<T>> T pickLarger(T a, T b) {
        return a.compareTo(b) >= 0 ? a : b;
    }

    public String summarize() {
        StringBuilder sb = new StringBuilder();
        for (Book book : books) {
            sb.append(book.describe()).append('\n');
        }
        String result = sb.toString().trim();
        System.out.println(result);
        return result;
    }

    public List<String> titles() {
        List<String> out = new ArrayList<>();
        books.forEach(book -> out.add(book.title()));
        return out;
    }

    public void forEachTitle(Consumer<String> action) {
        for (String title : titles()) {
            action.accept(title);
        }
    }

    public Runnable auditTask() {
        return new Runnable() {
            @Override
            public void run() {
                summarize();
            }
        };
    }

    public class Shelf {
        public int countFiction() {
            int count = 0;
            for (Book book : books) {
                if (book.isModern()) {
                    count++;
                }
            }
            return count;
        }
    }

    public static class Stats {
        public static double averageYear(List<Book> sample) {
            double total = 0;
            for (Book book : sample) {
                total += book.year();
            }
            return sample.isEmpty() ? 0 : total / sample.size();
        }
    }
}

interface Describable {
    String describe();

    default String shortLabel() {
        return describe().substring(0, 8);
    }
}

record Book(String title, int year) implements Describable {
    public String describe() {
        return title + " (" + year + ")";
    }

    public boolean isModern() {
        return year >= 2000;
    }
}
